### DISCUSSION TITLE
Once I get the proj_id, how could I get the system_matrix, or the projection matrix?

### DISCUSSION THREAD

**Discussion by leishaohu** [astra-toolbox/astra-toolbox#458] [Q&A] [answered]
import astra
import numpy as np

*** 设置2D成像参数
im = shepp_logan_phantom()
im = resize(im,(400,400))
vol_geom = astra.create_vol_geom(400,400)

proj_geom = astra.create_proj_geom('parallel', 1, 500, np.linspace(0,np.pi, 180))
proj_id = astra.create_projector('line', proj_geom, vol_geom)

*** 获取投影几何结构
data = astra.projector.weights_projection()

** 生成系统矩阵
system_matrix = astra.matrix.create(proj_id) 

** 获取系统矩阵的大小    
M, N = system_matrix.shape

** 打印系统矩阵的一些元素,可以看到值为0或1  
print(system_matrix[0,:10])
print(system_matrix[-1,:10])

I want the matrix:
*  matrix_id = astra_mex_projector('matrix', projector_id); *
* S = astra_mex_matrix('get', matrix_id);*

But I don't know how to create it.

**Comment by askorikov** [comment]
I'm not sure if I got you right, but take a look here: https://astra-toolbox.com/apiref/projector.html#astra.projector.matrix

**Reply by leishaohu** [reply]
![astra toolbox](https://github.com/astra-toolbox/astra-toolbox/assets/47586964/d540ca5a-3ae2-4128-8dac-da0b78a9147c)
But when I use astra.matrix.get(matrix_id), I get a mistake:
![Snipaste_2024-03-07_11-21-42](https://github.com/astra-toolbox/astra-toolbox/assets/47586964/ab73d9f9-330e-46e5-a52f-7f4e78b919f8)
I hope you can guide me。

**Comment by askorikov** [comment, marked as answer]
This was fixed by c597f99. However, this patch hasn't yet been shipped to conda, so you would either need to downgrade to NumPy 1.19 (and Python <= 3.9), build Astra yourself, apply the fix in your local copy or wait until the patch reaches conda packages.

**Reply by leishaohu** [reply]
Thanks a lot, I'll try it.

**Reply by leishaohu** [reply]
Can I know approximately when this package will be added to conda？

**Reply by askorikov** [reply]
Actually, it just got available on Linux, and Windows packages should get released in the coming weeks. You are looking for Astra Toolbox version 2.1.3

**Reply by leishaohu** [reply]
OK，thanks a lot
